src/main/java/com/fleotadezuta/youthprogrammanager/service/ParentService.java

```diff
@@ -0,0 +1,50 @@
+package com.fleotadezuta.youthprogrammanager.service;
+
+import com.fleotadezuta.youthprogrammanager.mapper.ParentMapper;
+import com.fleotadezuta.youthprogrammanager.model.ParentDto;
+import com.fleotadezuta.youthprogrammanager.persistence.repository.ParentRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Service
+@AllArgsConstructor
+public class ParentService {
+    private final ParentMapper parentMapper;
+    private final ParentRepository parentRepository;
+
+    public Flux<ParentDto> getAllParents() {
+        return parentRepository.findAll()
+                .map(parentMapper::fromParentDocumentToParentDto);
+    }
+
+    public Mono<ParentDto> getParentById(String id) {
+        return parentRepository.findById(id)
+                .map(parentMapper::fromParentDocumentToParentDto);
+    }
+
+    public Mono<ParentDto> addParent(ParentDto parentDto) {
+        var parentDoc = parentMapper.fromParentDtoToParentDocument(parentDto);
+        return parentRepository.save(parentDoc)
+                .map(parentMapper::fromParentDocumentToParentDto);
+    }
```

Comment:
Ezt is lehetne az updatehez hasonló reaktívabbá tenni.
Illetve kellene itt a service rétegben validáció a telefonszámokra (ezt akár egy külön private method-ba is teheted, mondjuk `validateParent`. Legyen a paraméter is meg a visszatérési érték is Mono és ellenőrizze, hogy nem üres-e a telefonszám lista és hogy valid-e minden telefonszám a listában (ehhez valószínűleg valami library fog kelleni)